src/referral.cairo

```diff
@@ -60,143 +76,191 @@ mod Referral {
 
     #[constructor]
     fn constructor(
+        ref self: ContractState,
         admin: ContractAddress,
         naming_addr: ContractAddress,
         eth_addr: ContractAddress,
         min_claim_amount: u256,
         share: u256
     ) {
-        initializer(:admin);
-        naming_contract::write(naming_addr);
-        eth_contract::write(eth_addr);
-        min_claim::write(min_claim_amount);
-        default_comm::write(share);
+        let mut ownable_state = Ownable::unsafe_new_contract_state();
```

Comment:
Yes we can already use it because it will compile to the same sierra, components is just a better syntax. However since they just got released, I will update it to this new syntax.